src/linear_algebra/char_poly/coeff.lean

```diff
@@ -160,3 +161,35 @@ begin
   rw [coeff_zero_eq_eval_zero, char_poly, eval_det, mat_poly_equiv_char_matrix, ← det_smul],
   simp
 end
+
+variables {p : ℕ} [fact p.prime]
+
+lemma zmod.char_poly_pow_card_of_nonempty [nonempty n] (M : matrix n n (zmod p)) :
```

Comment:
Can you generalise this proof from `zmod` to other (finite) fields? Or would that require Galois theory of finite fields as prereq?

Reply:
Maybe. I'd need to iterate the Frobenius in the proof at least, and I'm not sure how hard that'd be.

Reply:
I've succeeded at generalizing `zmod.expand_card` to finite fields, so maybe this won't be too bad?